docs/performance.md

```diff
@@ -0,0 +1,53 @@
+# Performance
+
+## Rerenders
+Duplicate rerenders of React components is a common
```

Comment:
I would try to be more balanced here. A re-render is not necessarily slow. In general, they are actually quite fast. Often the render method runs quickly and the diffing between current and previous result is not that slow. But some components are very slow to render (the whole MessageList for example)